diff --git a/go-ele/restapi/configure_lifty.go b/go-ele/restapi/configure_lifty.go
index 462c5b1..938ea9f 100644
--- a/go-ele/restapi/configure_lifty.go
+++ b/go-ele/restapi/configure_lifty.go
@@ -44,13 +44,13 @@ func configureAPI(api *operations.LiftyAPI) http.Handler {
 
 	api.CurrentFloorHandler = operations.CurrentFloorHandlerFunc(func(params operations.CurrentFloorParams) middleware.Responder {
 		f := v.Current(params.CarID)
-		body := operations.CurrentFloorOKBody{ID: &f.ID, Name: &f.Name}
+		body := &operations.CurrentFloorOKBody{ID: f.ID, Name: f.Name}
 		return operations.NewCurrentFloorOK().WithPayload(body)
 	})
 	api.FloorCountHandler = operations.FloorCountHandlerFunc(func(params operations.FloorCountParams) middleware.Responder {
 		// return middleware.NotImplemented("operation .FloorCount has not yet been implemented")
 		length := int32(len(v.Floors()))
-		return operations.NewFloorCountOK().WithPayload(operations.FloorCountOKBody{
+		return operations.NewFloorCountOK().WithPayload(&operations.FloorCountOKBody{
 			Count: &length,
 		})
 	})
@@ -81,7 +81,7 @@ func configureAPI(api *operations.LiftyAPI) http.Handler {
 	})
 	api.WelcomeHandler = operations.WelcomeHandlerFunc(func(params operations.WelcomeParams) middleware.Responder {
 		w := "Welcome to the Elevator Server"
-		return operations.NewWelcomeOK().WithPayload(operations.WelcomeOKBody{Msg: &w})
+		return operations.NewWelcomeOK().WithPayload(&operations.WelcomeOKBody{Msg: &w})
 	})
 
 	api.ServerShutdown = func() {
